R/rate0compare.R

```diff
@@ -0,0 +1,271 @@
+#' stratified  MH TEST
+#' @param Treatment treament indicator
+#' @param Response response indicator, 1 for responsder, 0 for non-responder
+#' @param TRT_ord treatment order. The first one is control arm and the second one is experiment arm. Default is NULL
+#' @param stratified Default is NULL for unstratified MN. If TRUE, stratified MN will be conducted
+#' @param Stratum stratum information
+#' @param delta delta under the null
+#' @param weight_schema weighting schema used in stratified MN method. "equal" for equal weighting, "ss" for sample size weighting, "cmh" for Cochran Mantel-Haenszel's weights. Default is "ss". 
+#' @param test a character string specifying the side of p-value, must be one of "right" (default), "left" or "two.sided".
+#' @param num the number of sections in the interval used in Bisection Method. Default is 100.
+#' @param eps the level of precision. Default is eps=1e-06.
+#' @param alpha pre-difined alpha level for Confidence Interval
+#' @export
+# library(haven)
+# adrs<-read_sas("//bardsar-test/mk7902-nsclc/prot008-ia1-dmc/dataanalysis/adrs.sas7bdat")
+# adsl<-read_sas("//bardsar-test/mk7902-nsclc/prot008-ia1-dmc/dataanalysis/adsl.sas7bdat")
+# adsl<-adsl[adsl$IA01FL=='Y'&adsl$TRT01PN !=7,c("USUBJID","TRT01P","STRATP")]
+# adrs<-adrs[adrs$PARAMCD=='BORCFIRC',]
+# adrs$response<-ifelse(adrs$AVALC %in% c('CR', 'PR'),1,0)
+# adrs<-adrs[,c("USUBJID","response")]
+# ana<-merge(adrs,adsl,by='USUBJID',all.y=TRUE)
+# 
+# 
+# Treatment<-ana$TRT01P
+# Response<-ana$response
+# Stratum<-ana$STRATP
+# library(dplyr)
+# library(tidyr)
+rate0compare <- function(Treatment, Response, Stratum, stratified=NULL, TRT_ord = c("Pembrolizumab+ Placebo","Pembrolizumab+ Ipilimumab"),

```

Comment:
I would suggest a function input as below. 

`formula = response ~ treatment | stratum` can be used to reduce parameters. 
 `data` can be used similar to `lm`
`TRT_ord` can be defined by requiring `treatment` to be a vector. 
`stratified ` is not required. As long as "stratum` is provided, it performs stratified test. 

Please use lower cases for all function and variable name. Please refer https://style.tidyverse.org/syntax.html

```suggestion
rate_compare <- function(formula, 
                                          data,
                                          delta = 0, 
                                          weight_schema= c('ss','equal','cmh'),
                                          test = c('two.sided', 'left', 'right'),
                                          alpha=0.05, 
                                          bisection = 100, 
                                          eps=1e-06)

```

Reply:
updated